### PR TITLE
app-content-pages: Add stories and RTL unit tests to publications components

### PR DESCRIPTION
**Pull request by goplayoutside3**
~~⚠️ Contains https://github.com/zooniverse/front-end-monorepo/pull/4643 please review that one first.~~

## Package
app-content-pages

## Describe your changes
- Added stories for PublicationsContainer, Publication, Category, and Project
- Converted unit tests to RTL for those four components
- Added translations keys for publication form instruction
- Tidied up prop-types
- Refactored Publication to handle edge case of missing url
- Changed the sidebar of the publications page to a `menu` rather than just a list of buttons
- Edited the mock publications data used for tests and stories (reduced the number of publications)

## How to Review

- Review app-content-pages's storybook. There should be stories for all publications components
- Run this branch locally and visit https://local.zooniverse.org:3000/about/publications?env=production to make sure it matches production https://www.zooniverse.org/about/publications

I won't be adding the translation keys to Lokalise right away. I'll wait until app-content-pages is fully refactored during maintenance week.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [x] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)

## General UX
- [x] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [x] The PR creator has described the reason for refactoring
- [x] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/app-content-pages/src/screens/Publications/Publications.js [hunk 1]
```diff
@@ -55,21 +50,22 @@ function Publications ({
   )
 
   const sidebar = (
-    <Box as='ul' gap='small'>
+    <Box as='menu' gap='small' name='Filter by category'>
```
Comment (eatyourgreens) [inline packages/app-content-pages/src/screens/Publications/Publications.js:53]: Menu is deprecated in HTML. I’d leave this as `<ul>` for screenreaders.

File: packages/app-content-pages/src/screens/Publications/Publications.js [hunk 2]
```diff
@@ -55,21 +50,22 @@ function Publications ({
...
-              onClick={filter.setActive}
-              plain
-            />
-          </Link>
-        </StyledLi>
+        <Link
+          href={filter.slug ? `#${filter.slug}` : ''}
+          key={filter.name}
+          passHref
+          role='menuitemcheckbox'
```
Comment (eatyourgreens) [inline packages/app-content-pages/src/screens/Publications/Publications.js:59]: I’d remove this role too, since the link here is being used as a link so we don’t want to override its role.

File: packages/app-content-pages/src/screens/Publications/Publications.js (hunk 1 above)
Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/Publications.js:53]: Sorry `<menuitem>` is deprecated in HTML, but I think `<ul>` might have better support than `<menu>` for marking up a list of links. See also #667.

File: packages/app-content-pages/src/screens/Publications/Publications.js (hunk 2 above)
Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/Publications.js:59]: These links also need to be wrapped in list items, in order for this to be valid HTML.

Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/Publications.js:59]: Seems like this PR is reverting some of the accessibility fixes from #3599.

File: packages/app-content-pages/src/screens/Publications/Publications.js (hunk 1 above)
Comment (goplayoutside3) [reply packages/app-content-pages/src/screens/Publications/Publications.js:53]: Okay will revert

File: packages/app-content-pages/src/screens/Publications/Publications.js (hunk 2 above)
Comment (goplayoutside3) [reply packages/app-content-pages/src/screens/Publications/Publications.js:59]: For clarity, do the links need to be wrapped in list items because their parent is `ul`? Or do they need to be wrapped in list items simply because they're links?

Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/Publications.js:59]: With the code here, it’s because the parent is [`<menu>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/menu), but the same rule is true for `<ul>` too. The children of a list element must be list items.

Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/Publications.js:59]: Here are some examples:
```html
<!-- valid list of links -->
<ul>
  <li><a href="#astronomy">Astronomy</a></li>
  <li><a href="#nature">Nature</a></li>
</ul>

<!-- valid menu (browsers parse <menu> as <ul>) -->
<menu>
  <li><a href="#astronomy">Astronomy</a></li>
  <li><a href="#nature">Nature</a></li>
</menu>

<!-- invalid menu (<a> can't be a direct child of <menu>) -->
<menu>
  <a href="#astronomy">Astronomy</a>
  <a href="#nature">Nature</a>
</menu>
```

Comment (goplayoutside3) [reply packages/app-content-pages/src/screens/Publications/Publications.js:59]: Thanks! I was a bit confused with this feature because it's a menu of links, but its UX is a filter feature.

Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/Publications.js]: With the previous implementation, which was a list of buttons (no links), it wasn’t clear that a blind reader would know the main content had changed after they pressed a button. 

Here, clicking a link navigates you to the corresponding heading, which will be announced.

Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/Publications.js]: I guess this pattern has the added bonus that you don’t lose your place if you refresh the page.

File: packages/app-content-pages/src/screens/Publications/PublicationsContainer.spec.js [hunk 3]
```diff
@@ -1,49 +1,56 @@
...
-      expect(componentWrapper.prop('data')).to.deep.equal(expectedResult)
-      expect(componentWrapper.prop('filters')[2].active).to.be.true()
-    })
+  it('should render all publications in data', function () {
+    render(
+      <RouterMock>
+        <DefaultStory />
+      </RouterMock>
+    )
+    const publications = screen.getAllByTestId('publication-test-element')
```
Comment (eatyourgreens) [inline packages/app-content-pages/src/screens/Publications/PublicationsContainer.spec.js:53]: IDs are unique in HTML, with each identifier only allowed to occur once in a document. It seems a bit counterintuitive to use test ID to group a set of elements here. That’s usually done in HTML with a CSS class.

File: packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js [hunk 4]
```diff
@@ -8,43 +8,56 @@ const StyledBox = styled(Box)`
...
-  const { authors, avatarSrc, className, title, url, year } = props
+function Publication({
+  authors = '',
+  avatarSrc = '',
+  className = '',
+  title = '',
+  url = '',
+  year = ''
+}) {
   const displayString = [title, authors, year].filter(v => v).join(', ')
```
Comment (eatyourgreens) [inline packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:33]: What is the filter doing here?

File: packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js [hunk 5]
```diff
@@ -8,43 +8,56 @@ const StyledBox = styled(Box)`
...
     <Box
       className={className}
+      data-testid='publication-test-element'
       direction='row'
       gap='small'
       margin={{ bottom: 'small' }}
     >
       <StyledBox height='50px' width='50px' flex={false}>
-        {!avatarSrc && Placeholder}
+        {!avatarSrc && <Placeholder id={displayString} />}
```
Comment (eatyourgreens) [inline packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:44]: I think this is an invalid identifier.

File: packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js [hunk 6]
```diff
@@ -8,43 +8,56 @@ const StyledBox = styled(Box)`
...
         {avatarSrc && (
           <Media
             height={50}
             src={avatarSrc}
-            placeholder={Placeholder}
+            placeholder={<Placeholder id={displayString} />}
```
Comment (eatyourgreens) [inline packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:49]: Possible invalid value for `id` here too. 

Browsers ignore invalid HTML. I’ll check what the rules are for identifiers.

**Review by eatyourgreens — APPROVED** [review]
Looks good. I’ve left minor comments about the HTML.

File: packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js (hunk 5 above)
Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:44]: >Technically, the value for an id attribute may contain any character, except [whitespace](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace) characters. However, to avoid inadvertent errors, only [ASCII](https://developer.mozilla.org/en-US/docs/Glossary/ASCII) letters, digits, '_', and '-' should be used, and the value for an id attribute should start with a letter.

https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id

**Comment by eatyourgreens** [comment]
ESLint warns of some problems during the build. Is it worth fixing them here, or in a separate PR? None of them seem difficult to fix.

```sh
./pages/_app.js
40:6  Warning: React Hook useEffect has a missing dependency: 'store?.user'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./src/api/publications/publications.js
61:1  Warning: Assign object to a variable before exporting as module default  import/no-anonymous-default-export

./src/api/team/team.js
9:1  Warning: Assign object to a variable before exporting as module default  import/no-anonymous-default-export

./src/screens/Publications/Publications.stories.js
4:1  Warning: Assign object to a variable before exporting as module default  import/no-anonymous-default-export

./src/screens/Publications/components/Category/Category.stories.js
3:1  Warning: Assign object to a variable before exporting as module default  import/no-anonymous-default-export

./src/screens/Publications/components/Project/Project.stories.js
3:1  Warning: Assign object to a variable before exporting as module default  import/no-anonymous-default-export

./src/screens/Publications/components/Publication/Publication.stories.js
3:1  Warning: Assign object to a variable before exporting as module default  import/no-anonymous-default-export

./src/shared/components/AboutHeader/AboutHeader.stories.js
3:1  Warning: Assign object to a variable before exporting as module default  import/no-anonymous-default-export

./src/shared/components/AuthModal/theme.js
3:1  Warning: Assign object to a variable before exporting as module default  import/no-anonymous-default-export
```

File: packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js (hunk 5 above)
Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:44]: The accessibility checker in the storybook catches invalid IDs. That's useful.

The AXE accessibility checker in the storybook, warning that `id` must be unique on the Publications page.

File: packages/app-content-pages/src/screens/Publications/Publications.js [hunk 7]
```diff
@@ -54,15 +53,16 @@ function Publications ({
   )
 
   const sidebar = (
-    <Box as='ul' gap='small'>
+    <Box as='ul' gap='small' name='Filter by category'>
```
Comment (eatyourgreens) [inline packages/app-content-pages/src/screens/Publications/Publications.js:56]: `gap` here is inserting invalid `div` elements inbetween the list items, resulting in this accessibility warning:
```
<ul> and <ol> must only directly contain <li>, <script> or <template> elements
```

This is a bug in Grommet, but we've worked around it before in https://github.com/zooniverse/front-end-monorepo/pull/1598.

Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/Publications.js:56]: Is `name` a property for Grommet flexboxes? I thought it would be `aria-label`. Or, better yet, wrap this list in a labeled navigation element:

```html
<nav aria-label='Filter by category'>
  <ul>
    <li><a href='#space'>Space</a></li>
    <li><a href='#nature'>Nature</a></li>
  </ul>
</nav>
```

"Filter by category" should also be localised here, so that it can be translated into other languages.

File: packages/app-content-pages/src/screens/Publications/PublicationsContainer.spec.js (hunk 3 above)
Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/PublicationsContainer.spec.js:53]: Not a blocking comment, by the way. This just struck me as odd.

Comment (goplayoutside3) [reply packages/app-content-pages/src/screens/Publications/PublicationsContainer.spec.js:53]: Definitely agreed that `id` is only allowed to occur once, and is that always true for `data-testid`? I was thinking of it like [data-](url).

I supposed I could use `className` here instead, but just wanted a confident way to count the number of publications rendered.

File: packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js (hunk 5 above)
Comment (goplayoutside3) [reply packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:44]: Thanks! I'll refactor this prop

File: packages/app-content-pages/src/screens/Publications/PublicationsContainer.spec.js (hunk 3 above)
Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/PublicationsContainer.spec.js:53]: I don’t think it’s a big deal. Just a bit strange.

File: packages/app-content-pages/src/screens/Publications/Publications.js (hunk 7 above)
Comment (goplayoutside3) [reply packages/app-content-pages/src/screens/Publications/Publications.js:56]: Good point about the localisation and `nav`, I'll refactor accordingly

File: packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js (hunk 4 above)
Comment (goplayoutside3) [reply packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:33]: Honestly I'm not sure 🤔

**Comment by goplayoutside3** [comment]
> ESLint warns of some problems during the build. Is it worth fixing them here

Sure! I'll take a look at the non-story ones

File: packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js (hunk 4 above)
Comment (goplayoutside3) [reply packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:33]: The filter helps in cases where one of the variables in array [title, authors, year] is missing. For example, if all three of those are missing the `displayString` is just two commas.

Comment (goplayoutside3) [reply packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:33]: In other words, we don't want to add commas to missing authors or year

Comment (eatyourgreens) [reply packages/app-content-pages/src/screens/Publications/components/Publication/Publication.js:33]:
```suggestion
  const displayString = [title, authors, year].filter(v => !!v).join(', ')
```

Converting `v` to a boolean might clarify that. I’m not sure.
